modules/parser/src/main/scala/com/gsk/kg/sparqlparser/Expression.scala

```diff
@@ -78,15 +78,21 @@ sealed trait StringVal extends StringLike {
   }
 }
 object StringVal {
-  final case class STRING(s: String)                   extends StringVal
-  final case class DT_STRING(s: String, tag: String)   extends StringVal
+  final case class STRING(s: String)                 extends StringVal
+  final case class DT_STRING(s: String, tag: String) extends StringVal
+  object DT_STRING {
+    def unApply(t: DT_STRING): String = s""""${t.s}"^^${t.tag}"""
+  }
   final case class LANG_STRING(s: String, tag: String) extends StringVal
-  final case class NUM(s: String)                      extends StringVal
-  final case class VARIABLE(s: String)                 extends StringVal
-  final case object GRAPH_VARIABLE                     extends StringVal { val s = "*g" }
-  final case class URIVAL(s: String)                   extends StringVal
-  final case class BLANK(s: String)                    extends StringVal
-  final case class BOOL(s: String)                     extends StringVal
+  object LANG_STRING {
+    def unApply(l: LANG_STRING): String = s""""${l.s}"${l.tag}"""
```

Comment:
Same here.  Also, shouldn't we use @ here?
```suggestion
    def unApply(l: LANG_STRING): String = s""""${l.s}"@${l.tag}"""
```

Reply:
Hmm, I'll take a look on this.